overlord/devicestate/devicestate.go

```diff
@@ -202,11 +203,51 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, _ snap.C
 	return nil
 }
 
+var (
+	// TODO: replace with gadget.IsCompatible() once we are ready
+	gadgetIsCompatible = func(current, new *gadget.Info) error { return nil }
+)
+
+func checkGadgetRemodelCompatible(st *state.State, snapInfo, curInfo *snap.Info, snapf snap.Container, flags snapstate.Flags, deviceCtx snapstate.DeviceContext) error {
+	if release.OnClassic {
+		return nil
+	}
+	if curInfo == nil || curInfo.GetType() != snap.TypeGadget {
```

Comment:
Note to self, when the check is running, curInfo is nil, since the snap we remodel to hasn't been fully installed yet. I'll be pushing a patch for this.